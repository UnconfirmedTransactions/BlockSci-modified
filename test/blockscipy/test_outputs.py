import pytest


def test_p2pkh_output_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2pkh-spend-2-tx"])
    output_regression(tx, regtest)


def test_p2sh_output_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2sh-spend-0-tx"])
    output_regression(tx, regtest)


@pytest.mark.btc
@pytest.mark.ltc
def test_p2wpkh_output_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2wpkh-spend-1-tx"])
    output_regression(tx, regtest)


@pytest.mark.btc
@pytest.mark.ltc
def test_p2wsh_output_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2wsh-spend-1-tx"])
    output_regression(tx, regtest)


def output_regression(tx, regtest):
    for out in tx.outputs:
        print(out.address, file=regtest)
        print(out.address_type, file=regtest)
        print(out.block, file=regtest)
        print(out.index, file=regtest)
        print(out.is_spent, file=regtest)
        print(out.spending_input, file=regtest)
        print(out.spending_tx, file=regtest)
        print(out.spending_tx_index, file=regtest)
        print(out.tx, file=regtest)
        print(out.tx_index, file=regtest)
        print(out.value, file=regtest)

def test_output_input_mapping(chain):
    for block in chain:
        for tx in block:
            for output in tx.outputs:
                if output.is_spent:
                    assert output == output.spending_input.spent_output
                    assert tx == output.spending_input.spent_tx
