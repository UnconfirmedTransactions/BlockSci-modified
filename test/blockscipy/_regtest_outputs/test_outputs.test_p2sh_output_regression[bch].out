ScriptHashAddress(2Mshjw3i5VcfqCm6FaJLxLSdSQhShKJYk6d, wrapped_address=unknown)
Pay to script hash
Block(tx_count=2, height=114, header_hash=6d94a93561acd2e5c3b5f845ff7d2b57a5829b73a603f831e313ffc49cde2a43, version=536870912, timestamp=1535763000, bits=545259519, nonce=0)
0
False
None
None
None
Tx(len(txins)=1, len(txouts)=2, size_bytes=223, block_height=114, tx_index=122)
122
100000000
PubkeyHashAddress(mjTkW3DjgyZck4KbiRusZsqTgaYTxdSz6z)
Pay to pubkey hash
Block(tx_count=2, height=114, header_hash=6d94a93561acd2e5c3b5f845ff7d2b57a5829b73a603f831e313ffc49cde2a43, version=536870912, timestamp=1535763000, bits=545259519, nonce=0)
1
False
None
None
None
Tx(len(txins)=1, len(txouts)=2, size_bytes=223, block_height=114, tx_index=122)
122
4899990000
